x/auth/client/testutil/suite.go

```diff
@@ -310,6 +314,120 @@ func (s *IntegrationTestSuite) TestCLIQueryTxCmd() {
 	}
 }
 
+func (s *IntegrationTestSuite) TestCLIQueryTxCmdByEvents() {
+	val := s.network.Validators[0]
+
+	account2, err := val.ClientCtx.Keyring.Key("newAccount2")
+	s.Require().NoError(err)
+
+	sendTokens := sdk.NewInt64Coin(s.cfg.BondDenom, 10)
+
+	// Send coins.
+	out, err := s.createBankMsg(
+		val, account2.GetAddress(),
+		sdk.NewCoins(sendTokens),
+	)
+	s.Require().NoError(err)
+	var txRes sdk.TxResponse
+	s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &txRes))
+	s.Require().NoError(s.network.WaitForNextBlock())
+
+	// Query the tx by hash to get the inner tx.
+	out, err = clitestutil.ExecTestCLICmd(val.ClientCtx, authcli.QueryTxCmd(), []string{txRes.TxHash, fmt.Sprintf("--%s=json", tmcli.OutputFlag)})
+	s.Require().NoError(err)
+	s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &txRes))
```

Comment:
the lines above a repeated in this file, can we move it to another function and reuse it?
Eg:
```
recipient, txRes, err := sendCoins(val, "newAccount2", 10)  // maybe we can add coins  to the return
```

Reply:
A function for sending coins already exists (`createBankMsg()`). I think putting the ExecTestCLICmd for `authcli.QueryTxCmd()` separately is okay.